### PR TITLE
Clarify restore hook includedResources when not specified

### PR DESCRIPTION
**Pull request by kaovilai**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [x] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by a-mccarthy — COMMENTED** [review]
thanks for submitting! editorially this LGTM, but i'd like to get a tech review as well. I'm a little confused about why leaving this field blank would apply to all resources, if only pods are supported. @zubron are you able to review this? or assign someone else who might have more time knowledge on the topic?  thanks!

**Comment by kaovilai** [comment]
> why leaving this field blank would apply to all resources, if only pods are supported.

I am just propagating the code comments from the api type. It will still apply to only pods right now due to supported resources being restricted to pods. If more resources are ever supported, leaving the field blank will result in more than just pods being supported.
https://github.com/vmware-tanzu/velero/blob/c65c17c5592eaed5800eaf66d98bb17b0377eac0/pkg/apis/velero/v1/restore.go#L48-L52

File: site/content/docs/main/api-types/restore.md [hunk 1]
```diff
@@ -88,8 +88,8 @@ spec:
       # Array of namespaces to which this hook does not apply. Optional.
       excludedNamespaces:
       - ns3
-      # Array of resources to which this hook applies. The only resource supported at this time is
-      # pods.
+      # Array of resources to which this hook applies. If empty, it applies to all supported resources.
```
Comment (jenting) [inline site/content/docs/main/api-types/restore.md:91]:
```suggestion
      # Array of resources to which this hook applies. If unspecified, the hook applies to all resources in the backup. Optional.
```

**Review by jenting — COMMENTED** [review]
@kaovilai 
In general LGTM. Just a nit suggestion about the wording.

**Review by jenting — APPROVED** [review]

**Review by a-mccarthy — APPROVED** [review]

**Pull request merged**